receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -5,28 +5,62 @@ package prometheusremotewritereceiver // import "github.com/open-telemetry/opent
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componentstatus"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver"
-	"go.uber.org/zap"
 )
 
-func newRemoteWriteReceiver(_ receiver.Settings, _ *Config, _ consumer.Metrics) (receiver.Metrics, error) {
-	panic("need implement")
+func newRemoteWriteReceiver(settings receiver.Settings, cfg *Config, nextConsumer consumer.Metrics) (receiver.Metrics, error) {
+	return &prometheusRemoteWriteReceiver{
+		settings:     settings,
+		nextConsumer: nextConsumer,
+		config:       cfg,
+		server: &http.Server{
+			ReadTimeout: 60 * time.Second,
+		},
+	}, nil
 }
 
-// nolint
 type prometheusRemoteWriteReceiver struct {
-	settings     receiver.Settings //nolint:unused
-	host         component.Host    //nolint:unused
-	nextConsumer consumer.Metrics  //nolint:unused
+	settings     receiver.Settings
+	nextConsumer consumer.Metrics
 
-	config *Config     //nolint:unused
-	logger *zap.Logger //nolint:unused
+	config *Config
+	server *http.Server
 }
 
-// nolint
-func (prwc *prometheusRemoteWriteReceiver) Start(_ context.Context, _ component.Host) error {
-	panic("need implement")
+func (prw *prometheusRemoteWriteReceiver) Start(ctx context.Context, host component.Host) error {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/api/v1/write", prw.handlePRWv2)
+	var err error
+
+	prw.server, err = prw.config.ToServer(ctx, host, prw.settings.TelemetrySettings, mux)
+	if err != nil {
+		return fmt.Errorf("failed to create server definition: %w", err)
+	}
+	listener, err := prw.config.ToListener(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create prometheus remote-write listener: %w", err)
+	}
+
+	go func() {
+		if err := prw.server.Serve(listener); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			componentstatus.ReportStatus(host, componentstatus.NewFatalErrorEvent(fmt.Errorf("error starting prometheus remote-write receiver: %w", err)))
+		}
+	}()
+	return nil
+}
+
+func (prw *prometheusRemoteWriteReceiver) Shutdown(ctx context.Context) error {
+	return prw.server.Shutdown(ctx)
```

Comment:
We probably need to check to make sure the server is not nil (e.g. if ToServer returns an error)